pkgs/sdk/server/src/Internal/BigSegments/BigSegmentStoreWrapper.cs

```diff
@@ -119,11 +121,7 @@ internal BigSegmentStoreStatus GetStatus()
             {
                 _lock.ExitReadLock();
             }
-            if (ret.HasValue)
-            {
-                return ret.Value;
-            }
-            return AsyncUtils.WaitSafely(() => PollStoreAndUpdateStatusAsync());
+            return ret ?? _initialPoll.GetAwaiter().GetResult();
```

Comment:
If there is no value, then that first poll has not completed. Wait for it to complete.

Reply:
Is it possible that the initial task is still executing by the time the first repeating task runs?  I suppose it may be extremely unlikely given the real world values of `StatusPollInterval`.

Reply:
It theoretically is possible. I did consider if there was a way to use the most recent task, but the complexity didn't seem worth it. If at any point a repeating task does complete, then we would ideally hit the first code path subsequently.